### PR TITLE
debug-lineage

### PR DESCRIPTION
**Pull request by sundarshankar89**

### CODE REVIEW

File: core/src/main/scala/com/databricks/labs/remorph/Main.scala [hunk 1]
```diff
@@ -36,6 +38,15 @@ object Main extends App with ApplicationContext {
         case nok: KoResult =>
           prettyPrinter(nok)
       }
+
+    case Payload("debug-lineage", args) =>
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/Main.scala:42]: needs changes to `cli.py` and `labs.yml` for this to work.

File: core/src/main/scala/com/databricks/labs/remorph/graph/TableGraph.scala [hunk 2]
```diff
@@ -13,6 +13,9 @@ class TableGraph(parser: PlanParser[_]) extends DependencyGraph with LazyLogging
   private val nodes = scala.collection.mutable.Set.empty[Node]
   private val edges = scala.collection.mutable.Set.empty[Edge]
 
+  def getNodes: Set[Node] = nodes.toSet
+  def getEdges: Set[Edge] = edges.toSet
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/graph/TableGraph.scala:17]:
```suggestion
  private[graph] val nodes = scala.collection.mutable.Set.empty[Node]
  private[graph] val edges = scala.collection.mutable.Set.empty[Edge]
```

create `GraphvizGenerator` class in the same package as `TableGraph` and use these package-private properties.

File: core/src/main/scala/com/databricks/labs/remorph/ApplicationContext.scala [hunk 3]
```diff
@@ -70,4 +72,23 @@ trait ApplicationContext {
...
+    tableGraph.getEdges.foreach { edge =>
+      edge.to.foreach { toNode =>
+        dotContent.append(s"""  "${edge.from.tableDefinition.table}" -> "${toNode.tableDefinition.table}";\n""")
+      }
+    }
+
+    dotContent.append("}\n")
+
+    Files.write(Paths.get(dstFile.getPath), dotContent.toString().getBytes)
+  }
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/ApplicationContext.scala:93]:
```suggestion
```

only declarative and dependency-binding code is allowed in `ApplicationContext`.

File: core/src/main/scala/com/databricks/labs/remorph/Main.scala [hunk 4]
```diff
@@ -36,6 +38,15 @@ object Main extends App with ApplicationContext {
...
+      val dst = new File(args("dst"))
+      val queryHistory = folderQueryHistoryProvider(args("src")).history()
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/Main.scala:44]: stack this PR on top of https://github.com/databrickslabs/remorph/pull/1051 - there's a fundamental change how we treat the query history.

**Review by nfx — CHANGES_REQUESTED** [review]

**Comment by sundarshankar89** [comment]
Closing this for now

**Pull request closed without merging**